tests/core/engine_adapter/test_integration.py

```diff
@@ -1364,6 +1364,7 @@ def test_scd_type_2_by_time(ctx: TestContext):
         execution_time="2023-01-01 00:00:00",
         updated_at_as_valid_from=False,
         columns_to_types=input_schema,
+        truncate=True,
```

Comment:
This properly mimics how the adapter will be used by truncating on first load and then not truncating on follow up load.